libknet/logging.c

```diff
@@ -216,12 +218,15 @@ void log_msg(knet_handle_t knet_h, uint8_t subsystem, uint8_t msglevel,
 
 	/*
 	 * most logging calls will take place with locking in place.
-	 * if we get an EINVAL and locking is initialized, then
+	 * if we get an EAGAIN and locking is initialized, then
 	 * we are getting a real error and we need to stop
```

Comment:
The starting "most" is rather hazy for documentation. Also, we can get any kind of error only if locking is initialized, so that constraint is confusing. But if we introduce `unlocked_log_msg()`, this whole comment can go.